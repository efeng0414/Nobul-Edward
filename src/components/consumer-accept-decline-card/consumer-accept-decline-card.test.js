import React from "react";
import renderer from "react-test-renderer";
import { IntlProvider } from "react-intl";

import ConsumerAcceptDeclineCardWrapped from "./index";
import ConsumerProposalDetailsButtonContainer from "../consumer-proposal-details-button-container";

describe("ConsumerAcceptDeclineCard", () => {
  const {
    WrappedComponent: ConsumerAcceptDeclineCard
  } = ConsumerAcceptDeclineCardWrapped;

  const componentProps = {
    intl: global.getIntl()
  };

  const componentJSX = <ConsumerAcceptDeclineCard {...componentProps} />;

  describe("Component Snapshot", () => {
    const snapshotJSX = (
      <IntlProvider {...global.intl}>{componentJSX}</IntlProvider>
    );

    const componentSnapshot = renderer.create(snapshotJSX).toJSON();
    it("should matched stored snapshot", () => {
      expect(componentSnapshot).toMatchSnapshot();
    });
  });

  describe("Render method", () => {
    const wrapper = global.Enzyme.mount(componentJSX);
    it("should render the card header ", () => {
      expect(wrapper.find(".consumer-card__header").length).toEqual(1);
    });

    it("should render the card body ", () => {
      expect(wrapper.find(".consumer-card__body").length).toEqual(1);
    });

    it("should render the child component containing buttons", () => {
      expect(
        wrapper.find(ConsumerProposalDetailsButtonContainer).length
      ).toEqual(1);
    });
  });
});
