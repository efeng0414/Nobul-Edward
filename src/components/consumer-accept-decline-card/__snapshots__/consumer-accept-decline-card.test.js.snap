// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsumerAcceptDeclineCard Component Snapshot should matched stored snapshot 1`] = `
<div
  className="consumer-card"
>
  <div
    className="consumer-card__header"
  >
    <div
      className="consumer-card__title"
    >
      Do you want to accept the agent's offer?
    </div>
  </div>
  <div
    className="consumer-card__body"
  >
    <div
      className="consumer-card__text-container"
    >
      <div
        className="consumer-card__text"
      >
        By clicking ACCEPT, I agree to schedule a meeting with this real estate agent based on their proposed offer. I acknowledge that doing so will dismiss all other offers(if applicable); I understand that the relationship between myself and the real estate agent is not binding until a Buyer Representation Agreement or Listing Agreement is signed by both parties.
      </div>
    </div>
    <div
      className="consumer-card__buttons"
    >
      <div
        className="accept-decline-buttons"
      >
        <div
          className="accept-decline-buttons__accept"
        >
          <button
            className="ant-btn decline-button ant-btn-lg"
            onClick={[Function]}
            type="button"
          >
            <span>
              Decline
            </span>
          </button>
        </div>
        <div
          className="accept-decline-buttons__decline"
        >
          <button
            className="ant-btn accept-button ant-btn-primary ant-btn-lg"
            onClick={[Function]}
            type="button"
          >
            <span>
              Accept
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
